File: internal/repo/models/case_template.go
package models

import "time"

const TableNameCaseTemplate = "case_template"

type CaseTemplate struct {
	ID          uint      `json:"id" gorm:"primaryKey;autoIncrement;comment:主键ID"`
	Name        string    `json:"name" gorm:"size:255;not null;uniqueIndex;comment:模板名称"`
	Content     string    `json:"content" gorm:"type:text;not null;comment:模板内容"`
	IsDefault   bool      `json:"is_default" gorm:"default:false;comment:是否为默认模板"`
	IsSystem    bool      `json:"is_system" gorm:"default:false;comment:是否为系统模板(不可删除)"`
	CreatedAt   time.Time `json:"created_at" gorm:"autoCreateTime;comment:创建时间"`
	UpdatedAt   time.Time `json:"updated_at" gorm:"autoUpdateTime;comment:更新时间"`
}

func (*CaseTemplate) TableName() string {
	return TableNameCaseTemplate
}

File: backend/app/models/case_template.py
"""用例模板模型"""
from datetime import datetime
from sqlalchemy import String, Integer, Boolean, DateTime, ForeignKey, JSON
from sqlalchemy.orm import Mapped, mapped_column
from app.core.database import Base


class CaseTemplate(Base):
    __tablename__ = "case_template"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    name: Mapped[str] = mapped_column(String(255), nullable=False, comment="模板名称")
    fields: Mapped[list] = mapped_column(JSON, nullable=False, comment="字段列表 [{'name': '字段名', 'description': '字段描述'}]")
    is_default: Mapped[bool] = mapped_column(Boolean, default=False, comment="是否默认模板")
    is_system: Mapped[bool] = mapped_column(Boolean, default=False, comment="是否系统模板")
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)
    updated_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
